physics/radiation_surface.f

```diff
@@ -956,26 +956,22 @@ subroutine setemis                                                &
           if ( icy(i) ) then                !-- ice emissivity
 
           !-- complete or fractional ice
-            if (lsm == lsm_noahmp) then
-              if (.not. cplice .or. lakefrac(i) > f_zero) then
-                if (sncovr_ice(i) > f_zero) then
-                  sfcemis_ice = emsref(7) * (f_one-sncovr_ice(i))         &
+            if (semis_ice(i) > f_zero) then
+              sfcemis_ice = semis_ice(i) ! output from RUC lsm or CICE (with snow effect)
+            else
+            !-- compute ice emissivity with snow effect for uncoupled lake or sea ice
+              if (sncovr_ice(i) > f_zero) then 
+                sfcemis_ice = emsref(7) * (f_one-sncovr_ice(i))         &
      &                        + emsref(8) * sncovr_ice(i)
-                elseif (snowf(i) > f_zero) then
-                  asnow = 0.02*snowf(i)
-                  argh  = min(0.50, max(.025,0.01*zorlf(i)))
-                  hrgh  = min(f_one,max(0.20,1.0577-1.1538e-3*hprif(i)))
-                  fsno  = asnow / (argh + asnow) * hrgh
-                  fsnoi = min(f_one, fsno / (fraci(i)+fracl(i)))
-                  sfcemis_ice = emsref(7)*(f_one-fsnoi) + emsref(8)*fsnoi
-                endif
-                semis_ice(i) = sfcemis_ice
-              else
-                sfcemis_ice = semis_ice(i) ! output from CICE
+              elseif (snowf(i) > f_zero) then
+                asnow = 0.02*snowf(i)
+                argh  = min(0.50, max(.025,0.01*zorlf(i)))
+                hrgh  = min(f_one,max(0.20,1.0577-1.1538e-3*hprif(i)))
+                fsno  = asnow / (argh + asnow) * hrgh
+                fsnoi = min(f_one, fsno / (fraci(i)+fracl(i)))
+                sfcemis_ice = emsref(7)*(f_one-fsnoi) + emsref(8)*fsnoi
```

Comment:
Tanya,
   There is one problem here.  Note that in my version I save "sfcemis_ice" into semis_ice(i), as it is needed elsewhere.  When we do this, it could potentially go through a different path in the above "if" loop when it is called another time.
Moorthi

Reply:
Moorthi,
I have changed the if statement to use semis_ice(i) with RUC lsm and CICE (for now other ice schemes or lake models do not provide semis_ice), and compute it in all other scenarios. Will this work now?

Reply:
Tanya,
    It won't work.  I think my original version is fine.  The reason it won't work is because, when it is coupled to CICE6, it only takes care of sea-ice, not lake ice.  That is the reason I was adding "lakefrac" in the logic so that if it is coupled and not RUClsm, then to compute emissivity for the lake ice the default way. 
Moorthi

Reply:
Ok, let's go with your version. Thank you for merging RUC LSM clean-up version into your PR.

Reply:
@SMoorthi-emc I forgot to tell you that one thing will not work in your version with RUC LSM. The RUC ice model is turned on for ice on the ocean and lakes only when uncoupled from CICE or the lake model. Thus, if coupled to the lake model semis_ice(i) is not defined unless it is computed in the lake model (I think it is not the case for Flake). Therefore, with RUC LSM the "standard" method to compute semis_ice from snow depth should be applied it the lake model is turned on.

Reply:
Moorthi, Yes to both questions. And yes, RUC LSM will calculate lake ice emissivity only if no lake model is used. Thank you!